PATCHES.md

```diff
@@ -1,20 +1,28 @@
 I use [Hopper Disassembler](http://www.hopperapp.com), it's available for OS X and Linux. You can use trial version that's is limited to 30 minutes which is enough.
 
+You can use Hopper to find the patches for all versions of DF. So you can run Hopper on OSX or Linux and use it to disassemble all of OSX, Windows and Linux versions.
+
 Note: usually addresses don't change much, and usually they increase. So if some address you found is completely different from the address in previous version, better check again if it's correct.
 
+The first time you do the patching on a new version it's recommended to first follow the instructions for a previous version that is already completed. That means you already have the found addresses which you can use to verify that you are understanding and following the instructions correctly. You can then refer back to the known-good version when searching in the new version, which can be very helpful when the new version has changed enough that it doesn't quite match the patterns described below.
+
 1. To find the address for `A_LOAD_MULTI_PDIM`:
 
 	**On Windows** find a function that references string "`Tileset not found`".
 	**On OS X** find a function that calls `IMG_Load` in its very beginning, and next `SDL_SetAlpha`, `SDL_CreateRGBSurface`, and `SDL_SetAlpha` again.
 	**On Linux** not required.
 
-2. Go to the address of vtable for `viewscreen_dwarfmodest` (from symbols.xml), you need *third* DWORD at that address, it's the address of viewscreen's `render()` method. Go to that function.
+2. Go to the address of vtable for `viewscreen_dwarfmodest` (from symbols.xml), you need *third* DWORD at that address, it's the address of viewscreen's `render()` method. Go to that function - note that on Windows, you will jump into the middle of a function, not the beginning.
```

Comment:
This is the process I followed, with screenshots from  40.24:

  <vtable-address name='viewscreen_dwarfmodest' value='0xdb5a44'/>

Go to 0xdb5a44 and see this in hex editor:  http://i.imgur.com/Nd40QM8.png

Instructions said go to "third DWORD", which, when reversed, is 00641E40 (highlighted in above screenshot)

Go to 00641E40, and see this: http://i.imgur.com/stuo5Mc.png

I've highlighted here the call to dwarfmode_render_main, which appears where you said it would, shortly after the point I jumped in, which is not the beginning of the function.

Looking into it more just now, I noticed that the beginning of this function I jumped into is the *first* DWORD after viewmode_dwarfmodest, ie its 00641540 which you can see in the first screenshot is the highlighted dword, at the address where viewmode_dwarfmodest takes us.

So the instructions could say "On Windows, you need the *first* DWORD at that address.."

However if you did go to the first DWORD, to the beginning of this function, it wouldn't then be the first call that goes to dwarfmode_render_main - in fact it's miles down in the function from that point (beginning of function is 00641540, the call to dwarfmode_render_main is at 00641e71).  So if we go to the beginning of the function on Windows, the rest of the instructions don't match.

I just checked further in that function to see if maybe there was another call to dwarfmode_render_main near the top, but there isn't.  So we do need to jump into the middle at the lower position.    And I've definitely got dwarfmode_render_main right because it matches your following instructions, and identifies render_map.

Is that all OK or did I do something wrong and just coincidentally end up at the right place? :)